README

```diff
@@ -367,31 +370,56 @@ throw new RuntimeException("Authentication failed");
 link:https://developers.yubico.com/java-webauthn-server/JavaDoc/webauthn-server-core/2.4.1/com/yubico/webauthn/AssertionResult.html[`AssertionResult`].
-Most importantly, you should update the signature counter. That might look something like this:
+Most importantly you should update the signature counter, and the backup state flag if you use it. That might look something like this:
 
 [source,java]
 ----------
 updateCredential(              // Some database access method of your own design
   "alice",                     // Query by username or other appropriate user identifier
   result.getCredentialId(),    // Query by credential ID of the credential used
-  result.signatureCount(),     // Set new signature counter value
+  result.getSignatureCount(),  // Set new signature counter value
+  result.isBackedUp(),         // Set new backup state flag
   Clock.systemUTC().instant()  // Set time of last use (now)
 );
 ----------
 
 Then do whatever else you need - for example, initiate a user session.
 
 
-=== 5. Passwordless, username-less authentication
+=== 5. Optional features: passkeys, multi-factor, backup state
 
-WebAuthn supports passwordless multi-factor authentication via on-authenticator
-https://www.w3.org/TR/webauthn-2/#user-verification[user verification],
-and username-less authentication via
-https://www.w3.org/TR/webauthn-2/#discoverable-credential[discoverable credentials]
-(sometimes the term "passwordless" is used to mean the combination of both, but
-here the two are treated separately).
+WebAuthn supports a number of additional features beyond the basics:
 
-Discoverable credentials must be enabled at registration time by setting the
+- <<passkeys,Passkeys>>: passwordless, username-less authentication.
+  link:https://passkey.org[Try it on passkey.org!]
+- <<user-verification,User verification>>: passwordless, streamlined multi-factor authentication.
+- <<autofill-ui,Autofill UI>>: Unintrusive passkey integration in traditional login forms.
+- <<credential-backup-state,Credential backup state>>: hints on how vulnerable the user is to authenticator loss.
+
+
+[#passkeys]
+==== Passkeys: passwordless, username-less authentication
+
+A https://passkeys.dev/[passkey] is a WebAuthn credential that can simultaneously both _identify_ and _authenticate_ the user.
+This is also called a link:https://www.w3.org/TR/2021/REC-webauthn-2-20210408/#discoverable-credential[discoverable credential].
+By default, credentials are created non-discoverable, which means the server
+must list them in the
+https://www.w3.org/TR/webauthn/#dom-publickeycredentialrequestoptions-allowcredentials[`allowCredentials`]
+parameter before the user can use them to authenticate.
+This is typically because the credential private key is not stored within the authenticator,
+but instead encoded into one of the credential IDs in `allowCredentials`.
+This way even a small hardware authenticator can have an unlimited credential capacity,
```

Comment:
I would put comma after "This way" and remove it from after "capacity". If you think the sentence would become too long, I would structure it to something like: ... credential capacity. However, there is still the drawback that...

Reply:
Thanks, but I feel like it flows better the way it is.